File: 第二章_排序/排序(3)/有序数组合并练习题/Merge.cpp
class Merge {
public:
    int* mergeAB(int* A, int* B, int n, int m) {
        int end=n+m-1;
		int j=m-1;
		int i=n-1;
		while(j>=0&&i>=0){
			if(A[i]>B[j]){
				A[end--]=A[i--];
			}else{
				A[end--]=B[j--];
			}
		}

		while(j>=0){
			A[end--]=B[j--];
		}

		return A;
    }
};

File: 第二章_排序/排序(3)/有序数组合并练习题/Merge.java
import java.util.*;
 
public class Merge {
    public int[] mergeAB(int[] A, int[] B, int n, int m) {
        int k=n+m-1;
        int i=n-1,j=m-1;
        while (i>=0&&j>=0){
            if (A[i]>B[j]){
                A[k--]=A[i--];
            }else {
                A[k--]=B[j--];
            }
        }
        while(j>=0){
            A[k--]=B[j--];
        }
        return A;
    }
}